File: Week 1 - C/ProblemSet1/MarioMore/mario.c
/*
https://cs50.harvard.edu/x/2023/psets/1/mario/more/
*/

#include <cs50.h>
#include <stdio.h>

int main(void)
{
    // Declare height variable
    int height;

    // Get user input
    do
    {
        height = get_int("Height: ");
    }
    while (height < 1 || height > 8);

    // Outer loop / Switch to the next line of the pyramid after printing the spaces and blocks
    for (int line = 0; line < height; line++)
    {
        // Print the spaces
        for (int space = height - 1; space > line; space--)
        {
            printf(" ");
        }

        // Print the left blocks
        for (int left_blocks = 0; left_blocks <= line; left_blocks++)
        {
            printf("#");
        }

        // Print the two spaces between the blocks
        printf("  ");

        // Print the right blocks
        for (int right_blocks = 0; right_blocks <= line; right_blocks++)
        {
            printf("#");
        }

        // Go to the next line
        printf("\n");
    }
}

File: Week 6 - Python/ProblemSet6/SentimentalMarioLess/mario.py
"""
https://cs50.harvard.edu/x/2023/psets/6/mario/less/
"""

def get_height():
    """Returns the height that was given by the user."""

    while True:
        try:
            height = int(input("Height: "))

            if height > 0 and height < 9:
                return height

        except ValueError as error:
            print(error)


def print_pyramid(height):
    """Prints out the pyramid."""

    for line in range(height):
        print(" " * (height - line - 1), end="")
        print("#" * (line + 1))


# Call the function that prints the pyramid which takes as argument the function that returns the height.
print_pyramid(get_height())
